nntrainer/src/fc_layer.cpp

```diff
@@ -158,4 +158,13 @@ Tensor FullyConnectedLayer::backwarding(Tensor derivative, int iteration) {
 
   return ret;
 }
+
+void FullyConnectedLayer::print(std::ostream &out) const {
+  Layer::print(out);
+
+  out << "unit: " << unit << std::endl << "= weight info =" << std::endl;
+
+  weight.print(out);
```

Comment:
is it possible to add a mechanism to restrict the level of information?
This will be useful for the C-API as we don't expose tensor level information (abstracted by layer) in the API, and printing summary for tensors might not be a good choice.

Reply:
I think this must be done at least internally though, I will make PRs as follows:

- [ ] Add neuralnet print
- [ ] Add summary control

since I am gonna heavily rely on delegation, it would be more reasonable to review control level's altogether